src/utils/MaxVaultViewer.sol

```diff
@@ -0,0 +1,96 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity ^0.8.24;
+
+import {IVault} from "src/BaseVault.sol";
+import {IProvider} from "src/interface/IProvider.sol";
+import {IERC20Metadata, Initializable, Math, AccessControlUpgradeable} from "src/Common.sol";
+import {BaseVaultViewer} from "src/utils/BaseVaultViewer.sol";
+
+contract MaxVaultViewer is BaseVaultViewer, AccessControlUpgradeable {
+    bytes32 public constant UPDATER_ROLE = keccak256("UPDATER_ROLE");
+
+    error ZeroAddress();
+    error InvalidAssets();
+    error InvalidAssetAdd(address);
+    error InvalidAssetRemove(address);
+
+    struct AssetStorage {
+        mapping(address => bool) assets;
+        uint256 assetsLength;
+    }
+
+    function initialize(address vault_, address admin_) external initializer {
+        __AccessControl_init();
+        _grantRole(DEFAULT_ADMIN_ROLE, admin_);
+
+        _getViewerStorage().vault = vault_;
+    }
+
+    function getStrategies() public view returns (AssetInfo[] memory assetsInfo) {
+        IVault vault = IVault(_getViewerStorage().vault);
+
+        address[] memory allAssets = vault.getAssets();
+
+        uint256 assetsLength = _getAssetStorage().assetsLength;
+        if (allAssets.length < assetsLength) revert InvalidAssets();
+
+        uint256 strategiesLength = allAssets.length - assetsLength;
+
+        address[] memory strategies = new address[](strategiesLength);
+        uint256[] memory balances = new uint256[](strategiesLength);
+
+        for (uint256 i = 0; i < strategiesLength; ++i) {
+            if (_getAssetStorage().assets[allAssets[i]]) continue;
+            strategies[i] = allAssets[i];
```

Comment:
this now works the opposite way it should work

the strategies are the things that are *not* underyling assets

Reply:
@danoctavian Here its exactly as you described it, strategies are indeed NOT underlying assets. 

If (asset is underlying) continue:
Else add to strategies lost

Reply:
fixed in [df7ca37](https://github.com/yieldnest/yieldnest-vault/pull/54/commits/df7ca37f9c0167de7115697aedeae9efadd02ba6)